rascaline/src/math/splines.rs

```diff
@@ -35,18 +35,18 @@ pub struct SplineParameters {
 
 /// A single control point/knot in the Hermit cubic spline
 #[derive(Debug, Clone)]
-struct HermitSplinePoint<D: ndarray::Dimension> {
+pub struct HermitSplinePoint<D: ndarray::Dimension> {
     /// Position of the point
-    position: f64,
+    pub position: f64,
     /// Value of the function to interpolate at the position
-    value: Array<f64, D>,
+    pub value: Array<f64, D>,
     /// Derivative of the function to interpolate at the position
-    derivative: Array<f64, D>,
+    pub derivative: Array<f64, D>,
 }
 
 impl<D: ndarray::Dimension> HermitCubicSpline<D> {
     #[allow(clippy::float_cmp)]
-    fn new(
+    pub fn new(
```

Comment:
This should not be public, since this is not something users of rascaline as a library should be able to use. It can be made public for other modules in rascaline with `pub(crate)` though.

A nicer solution might be to add a new public function that takes `Vec<f64>` for positions, and two `Vec<Array<f64, D>>` for values/gradient and move the validation (checking shapes, checking nothing is NaN/inf, …) inside this new function.